pynumdiff/utils/utility.py

```diff
@@ -51,17 +52,20 @@ def estimate_integration_constant(x, x_hat, M=6):
         mean absolute deviation of residuals, so scatter can be calculated and used to normalize without being
         thrown off by outliers. The default is intended to capture the idea of "six sigma": Assuming Gaussian
         :code:`x - xhat` errors, the portion of inliers beyond the Huber loss' transition is only about 1.97e-9.
+    :param int axis: data dimension along which integration was performed
 
-    :return: **integration constant** (float) -- initial condition that best aligns x_hat with x
+    :return: **integration constant** (float or np.array[float]) -- initial condition(s) to best align
+             :math:`\\mathbf{\\hat{x}}` with :math:`\\mathbf{x}`
     """
-    sigma = median_abs_deviation(x - x_hat, scale='normal') # M is in units of this robust scatter metric
-    if M == float('inf') or sigma < 1e-3: # If no scatter, then no outliers, so use L2
-        return np.mean(x - x_hat) # Solves the l2 distance minimization, argmin_{x0} ||x_hat + x0 - x||_2^2
+    s = list(x_hat.shape); s[axis] = 1; s = tuple(s) # proper shape for multidimensional integration constants
+    sigma = median_abs_deviation(x - x_hat, axis=axis, scale='normal') # M is in units of this robust scatter metric
+    if M == float('inf') or np.all(sigma < 1e-3): # If no scatter, then no outliers, so use L2
+        return np.mean(x - x_hat, axis=axis).reshape(s) # Solves the l2 distance minimization, argmin_c ||x_hat + c - x||_2^2
```

Comment:
Importantly, the return value has to be reshaped to `s` so the result can be `+=`ed to the multidimensional `x_hat` along the correct dimension.